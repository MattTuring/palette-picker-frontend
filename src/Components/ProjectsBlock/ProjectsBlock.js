import React from 'react';
import ProjectItem from '../ProjectItem/ProjectItem';

import AddProjectForm from '../AddProjectForm/AddProjectForm';


const ProjectsBlock = (props) => {
  const projectItems = props.projects.map((project, ind) => (
    <ProjectItem
      key={Date.now() + project.name + ind}
      project={project}
      {...props} />
  ));

  return (
    <section className="projects">
      <h2>Projects</h2>
      { projectItems }
      <AddProjectForm
        setProjects={props.setProjects}
        projects={props.projects}/>
    </section>
  )
};

export default ProjectsBlock;
