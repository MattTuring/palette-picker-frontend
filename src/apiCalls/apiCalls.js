export const getProjects = async () => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/projects';
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error('There was an error getting projects.');
  }
  const projects = await response.json();
  return projects
}

export const getPalettes = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}/palettes`
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error('There was an error getting palettes.');
  }
  const palettes = await response.json();
  return palettes
}

export const postProject = async (project) => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/projects';
  const options = {
    method: 'POST',
    body: JSON.stringify(project),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error posting project.');
  }

  const result = await response.json();
  return result;
}

export const postPalette = async (palette) => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/palettes';
  const options = {
    method: 'POST',
    body: JSON.stringify(palette),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error posting palette.');
  }

  const result = await response.json();
  return result;
}

export const updateProject = async (project, id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}`;
  const options = {
    method: 'PUT',
    body: JSON.stringify(project),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error updating project.');
  }

  const result = await response.json();
  return result;
}

export const updatePalette = async (palette, id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/palettes/${id}`;
  const options = {
    method: 'PUT',
    body: JSON.stringify(palette),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error updating palette.');
  }

  const result = await response.json();
  return result;
}

export const deleteProject = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}`;
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error deleting project.');
  }

  const result = await response.json();
  return result;
}

export const deletePalette = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/palettes/${id}`;
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error deleting palette.');
  }

  const result = await response.json();
  return result;
}
