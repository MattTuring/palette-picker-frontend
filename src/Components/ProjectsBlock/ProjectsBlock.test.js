import React from 'react';
import { shallow } from 'enzyme';
import ProjectsBlock from './ProjectsBlock';
import { deletePalette } from '../../apiCalls/apiCalls';

jest.mock('../../apiCalls/apiCalls');

describe('ProjectsBlock', () => {
  const setProjects = jest.fn();
  const mockProps = {
    projects: [{
      id: 1,
      name: 'Project'
    }],
    setProjects};

  let wrapper;

  beforeEach(() => {
    Date.now = jest.fn().mockImplementation(() => 1);

    wrapper = shallow(
      <ProjectsBlock {...mockProps} />
    );
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
