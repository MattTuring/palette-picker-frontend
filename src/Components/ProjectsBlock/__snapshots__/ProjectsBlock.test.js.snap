// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsBlock should match the snapshot with all data passed in correctly 1`] = `
<section
  className="projects"
>
  <h2>
    Projects
  </h2>
  <ProjectItem
    key="1Project0"
    project={
      Object {
        "id": 1,
        "name": "Project",
      }
    }
    projects={
      Array [
        Object {
          "id": 1,
          "name": "Project",
        },
      ]
    }
    setProjects={[MockFunction]}
  />

  <AddProjectForm
    projects={
      Array [
        Object {
          "id": 1,
          "name": "Project",
        },
      ]
    }
    setProjects={[MockFunction]}
  />

</section>
`;
